lib-injection/sitecustomize.py

```diff
@@ -79,29 +265,52 @@ def _inject():
-            #     now that the package is installed.
-            #   - Add the custom site-packages directory to PYTHONPATH to ensure the ddtrace package can be loaded
-            #   - Add the ddtrace bootstrap dir to the PYTHONPATH to achieve the same effect as ddtrace-run.
-            python_path = os.getenv("PYTHONPATH", "").split(os.pathsep)
-            if script_dir in python_path:
-                python_path.remove(script_dir)
-            python_path.insert(0, site_pkgs_path)
-            bootstrap_dir = os.path.abspath(os.path.dirname(ddtrace.bootstrap.sitecustomize.__file__))
-            python_path.insert(0, bootstrap_dir)
-            python_path = os.pathsep.join(python_path)
-            os.environ["PYTHONPATH"] = python_path
-
-            # Also insert the bootstrap dir in the path of the current python process.
-            sys.path.insert(0, bootstrap_dir)
-            _log("successfully configured ddtrace package, python path is %r" % os.environ["PYTHONPATH"])
+            try:
+                import ddtrace.bootstrap.sitecustomize
+
+                # Modify the PYTHONPATH for any subprocesses that might be spawned:
+                #   - Remove the PYTHONPATH entry used to bootstrap this installation as it's no longer necessary
+                #     now that the package is installed.
+                #   - Add the custom site-packages directory to PYTHONPATH to ensure the ddtrace package can be loaded
+                #   - Add the ddtrace bootstrap dir to the PYTHONPATH to achieve the same effect as ddtrace-run.
+                python_path = os.getenv("PYTHONPATH", "").split(os.pathsep)
+                if script_dir in python_path:
+                    python_path.remove(script_dir)
+                python_path.insert(0, site_pkgs_path)
+                bootstrap_dir = os.path.abspath(os.path.dirname(ddtrace.bootstrap.sitecustomize.__file__))
+                python_path.insert(0, bootstrap_dir)
+                python_path = os.pathsep.join(python_path)
+                os.environ["PYTHONPATH"] = python_path
+
+                # Also insert the bootstrap dir in the path of the current python process.
+                sys.path.insert(0, bootstrap_dir)
+                _log("successfully configured ddtrace package, python path is %r" % os.environ["PYTHONPATH"])
+                event = gen_telemetry_payload(
+                    [
+                        create_count_metric(
+                            "library_entrypoint.complete",
+                            [
+                                "injection_forced:" + str(runtime_incomp or integration_incomp).lower(),
+                            ],
+                        )
+                    ]
+                )
+                send_telemetry(event)
+            except Exception as e:
+                event = gen_telemetry_payload(
+                    [create_count_metric("library_entrypoint.error", ["error:" + type(e).__name__.lower()])]
```

Comment:
We should have these tags, only if available and relevant, otherwise use “NA”.
```
integration:<integration name where the error occurred>

integration_version: <integration_version>
```

Reply:
I'll follow up on this in a subsequent pull request

Reply:
@ZStriker19 do you have any recommendations on how to determine appropriate values for these tags from within this exception handler?